aikau/src/main/resources/alfresco/layout/HeightMixin.js

```diff
@@ -0,0 +1,129 @@
+       * This property allows the height of the sidebar to accommodate a "sticky" footer. The height is otherwise calculated as
+       * the height of the view port minus the top position of the DOM element having its height set. By setting this property 
+       * it is possible to also deduct the height of a sticky footer. 
+       * 
+       * @instance
+       * @type {number} 
+       * @default
+       */
+      footerHeight: 0,
+      
+      /**
+       * <p>This should be configured to indicate how the height of the widget should be calculated.
+       *   <ul>
+       *     <li>"AUTO" (the default) indicates that the height will be calculated to be the available space from the widgets position
+       * to the bottom of the screen (minus any [footerHeight]{@link module:alfresco/layout/HeightMixin#footerHeight}).<li>
+       *     <li>"DIALOG" indicates that the height should be taken from the available height of the dialog in which the widget
+       * is displayed.</li>
+       *     <li>Any negative number indicates that the "AUTO" height minus the supplied value will be used.</li>
+       *     <li>Any positive number indicates a fixed height (in pixels) that should be used for the height</li>
+       *   </ul>
+       * </p>
+       * 
+       * @instance
+       * @type {string|number}
+       * @default
+       */
+      heightMode: "AUTO",
+
+      /**
+       * Calculates the height of the supplied element based on the available space using the configured 
+       * [heightMode]{@link module:alfresco/layout/HeightMixin#heightMode} setting.
+       * 
+       * @instance
+       * @param {element} domNode The DOM element to calculate the height for.
+       */
+      calculateHeight: function alfresco_layout_HeightMixin__calculateHeight(domNode) {
+         var calculatedHeight;
+         if (domNode)
+         {
+            // Get the position of the DOM node and the available view port height...
+            var winBox = win.getBox();
+            var heightMode = this.heightMode;
+            if (heightMode === "DIALOG")
+            {
+               calculatedHeight = $(domNode).parentsUntil(".alfresco-dialog-AlfDialog").last().height();
+            }
+            else if (!heightMode || heightMode === "AUTO" || isNaN(heightMode))
+            {
+               // We're using JQuery here to get the offset as it has proved more reliable than either the Dojo margin box
+               // or native browser offsetTop options...
+               var offset = $(domNode).offset().top;
+               var availableHeight = winBox.h - offset - this.footerHeight;
+               calculatedHeight =  availableHeight;
+            }
+            else if (heightMode < 0)
+            {
+               // If the height mode is a number less than zero, then deduct that height from the available space.
+               calculatedHeight = winBox.h + heightMode; // NOTE: Not a mistake, remember adding a negative number substracts! :)
```

Comment:
This method doesn't follow the JSDoc explanation of what happens if you specify a negative number in heightMode. JSDoc says it'll be the same as AUTO, but with the heightMode subtracted, however the calculation for AUTO is different than just winBox.h